v2/test/mysql_test.go

```diff
@@ -277,7 +277,49 @@ func MySQL_User_CRUD(tc *testcommon.KubePerTestContext, server *mysql.FlexibleSe
 		2*time.Minute, // We expect this to pass pretty quickly
 	).Should(Succeed())
 
+	originalUser := user.DeepCopy()
+
+	// Confirm that we cannot change the user owner
+	old = user.DeepCopy()
+	user.Spec.Owner.Name = "adifferentowner"
+	err = tc.PatchAndExpectError(old, user)
+	tc.Expect(err).To(HaveOccurred())
+	tc.Expect(err.Error()).To(ContainSubstring("updating 'Owner.Name' is not allowed"))
+
+	// Confirm that we cannot change the user AzureName
+	user = originalUser.DeepCopy()
+	old = user.DeepCopy()
+	user.Spec.AzureName = "adifferentname"
+	err = tc.PatchAndExpectError(old, user)
+	tc.Expect(err).To(HaveOccurred())
+	tc.Expect(err.Error()).To(ContainSubstring("updating 'AzureName' is not allowed"))
+
+	// Confirm that we cannot change the user type from local to AAD
+	user = originalUser.DeepCopy()
+	old = user.DeepCopy()
+	user.Spec.LocalUser = nil
+	user.Spec.AADUser = &mysqlv1.AADUserSpec{
+		ServerAdminUsername: "someadminuser",
+	}
+	err = tc.PatchAndExpectError(old, user)
+	tc.Expect(err).To(HaveOccurred())
+	tc.Expect(err.Error()).To(ContainSubstring("cannot change from local user to AAD user"))
```

Comment:
Do we need the reverse test (cannot change from AAD user to local user) too?

Reply:
Not a bad idea, although it's tricky because right now we can't actually successfully create an AAD user in our CI subscription. I've added this to the AAD user test but it's skipped right now like all of the other bits requiring the ability to create an AAD user in our CI sub.